manifests/init.pp

```diff
@@ -193,7 +193,7 @@
   }
 
   # Whole solr configuration directory can be recursively overriden
-  if $solr::source_dir {
+  if $solr::source_dir != '' {
```

Comment:
I'd check here also for false values so:
if $solr::source_dir and $solr::source_dir != ''